akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -434,12 +412,12 @@ class TLSActor(
     if (tracing) log.debug("handshake finished")
     val session = engine.getSession
 
-    hostInfo.map(_._1) match {
-      case Some(hostname) if !hostnameVerifier.verify(hostname, session) ⇒
```

Comment:
I didn't find any documentation why we need to add this extra verification here. It seems that using `setEndpointIdentificationAlgorithm` should be enough to trigger the built in host name verification. Is it to support any extra verifications in addition to what `setEndpointIdentificationAlgorithm` provides?

This change might have been prompted by this post from @wsargent:

https://tersesystems.com/2014/03/23/fixing-hostname-verification/

Was it?

It might be that the arguments from that post are not valid any more now that we are on Java 8 which reliably supports host name verification but I'm not sure.

If we could get rid of this, we could simplify the interface of the TLSActor even more.